app/index.js

```diff
@@ -20,6 +20,18 @@ app.config = config;
 app.plugins = plugins;
 app.getWindows = () => new Set([...windowSet]); // return a clone
 
+// function to retrive the last focused winow in windowSet;
+// added to app object in order to expose it to plugins.
+app.getLastFocusedWindow = () => {
+  let lastWindow = null;
```

Comment:
Wow, reduce is perfect in this case, never used it, thank you!
However we can't do that because windowSet is a Set, not an array. But i could convert it to an array with `Array.from(windowSet);` :)
Now I'm going to sleep, but tomorrow I'll do that!